src/pages/Search/SearchPage.tsx

```diff
@@ -802,243 +828,97 @@ function SearchPage({route}: SearchPageProps) {
         return {count, total, currency};
     }, [areAllMatchingItemsSelected, metadata?.count, metadata?.currency, metadata?.total, selectedTransactions, selectedTransactionsKeys.length]);
 
-    if (shouldUseNarrowLayout) {
-        return (
-            <>
-                <DragAndDropProvider>
-                    {PDFValidationComponent}
-                    <SearchPageNarrow
+    const onSortPressedCallback = useCallback(() => {
+        setIsSorting(true);
+    }, []);
+
+    const scrollHandler = useCallback(
+        (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+            if (!e.nativeEvent.contentOffset.y) {
+                return;
+            }
+
+            saveScrollOffset(route, e.nativeEvent.contentOffset.y);
+        },
+        [saveScrollOffset, route],
+    );
+
+    return (
+        <>
+            <Animated.View style={[styles.flex1]}>
+                {shouldUseNarrowLayout ? (
+                    <DragAndDropProvider>
+                        {PDFValidationComponent}
+                        <SearchPageNarrow
+                            queryJSON={queryJSON}
+                            metadata={metadata}
+                            headerButtonsOptions={headerButtonsOptions}
+                            searchResults={searchResults}
+                            isMobileSelectionModeEnabled={isMobileSelectionModeEnabled}
+                            footerData={footerData}
+                            currentSelectedPolicyID={selectedPolicyIDs?.at(0)}
+                            currentSelectedReportID={selectedTransactionReportIDs?.at(0) ?? selectedReportIDs?.at(0)}
+                            confirmPayment={onBulkPaySelected}
+                            latestBankItems={latestBankItems}
+                        />
+                        <DragAndDropConsumer onDrop={initScanRequest}>
+                            <DropZoneUI
+                                icon={expensifyIcons.SmartScan}
+                                dropTitle={translate('dropzone.scanReceipts')}
+                                dropStyles={styles.receiptDropOverlay(true)}
+                                dropTextStyles={styles.receiptDropText}
+                                dropWrapperStyles={{marginBottom: variables.bottomTabHeight}}
+                                dashedBorderStyles={[styles.dropzoneArea, styles.easeInOpacityTransition, styles.activeDropzoneDashedBorder(theme.receiptDropBorderColorActive, true)]}
+                            />
+                        </DragAndDropConsumer>
+                        {ErrorModal}
+                    </DragAndDropProvider>
+                ) : (
+                    <SearchPageWide
```

Comment:
The 'wide' variant does not rely on any custom logic. Can we instead put its content directly in here to avoid drilling the props?